app/scripts/controllers/transactions/index.js

```diff
@@ -879,4 +822,53 @@ export default class TransactionController extends EventEmitter {
     this.memStore.updateState({ unapprovedTxs, currentNetworkTxList })
   }
 
+  _trackSwapsMetrics (txMeta) {
+    if (this._getParticipateInMetrics() && txMeta.swapMetaData) {
```

Comment:
We check if MetaMetrics are enabled here to save us from computing the metrics data. The `trackSegmentEvent` function also performs this check as a safety measure.